src/info/worker.ts

```diff
@@ -694,14 +644,89 @@ function activeGuidance(segments: SubSegment[][], tSegmentDuration: number,
         tLastChangePoint = Date.now();
       } else {
 
-        // move to end effector to the point
-        const coord = currentSubSegment.coordinates[currentSubSegmentPointIndex];
-        moveToPos(coord);
+        // move to the point
+                // to avoid setting a large force at once
+                // when moving from the end of one segment to the beginning of another
+                // we linear interpolate the # of points
+                // to avoid setting large forces
+                // except in the case of setting the force from home pos
+                // which needs a larger force to compensate
+                if (currentSubSegmentPointIndex == 0 && !atHomePos()) {
+
+                  // case where we need to transition from segment to segment
+                  switch (transition) {
+                      case Transition.GetPoints: {
+                          // interpolate point data
+                          const vEndEffector = new Vector(convPosEE.x, convPosEE.y);
+                          upSampled = upsample([vEndEffector, coord], 6500);
+                          tHoldTimeSegToSeg = Date.now();
+                          transition = Transition.Move;
+                          break;
+                      }
+                      case Transition.Move: {
+                          // if we are done then end
+                          if (idx >= upSampled.length - 1) {
+                              transition = Transition.Rest;
+                              idx = 0;
+                              break;
+                          }
+                          else {
+                              // force reduction as we approach the threshold
+                              let k = 3;
+                              let coeff = 1;
+                              if (diff.mag() < 0.025) {
+                                  let unit = diff.unit();
+                                  const multiplier = 1 / (diff.mag() + 0.9);
+                                  const coeff = Math.min(1, 1 - ((unit.x * unit.y) * multiplier * multiplier));
+                              }
+                              // move to new point with the WaitTime refresh rate
+                              if (Date.now() - tHoldTimeSegToSeg > tWaitTime) {
```

Comment:
tWaitTime is undefined?

Reply:
this should be defined at the top now as a const